hal/armv7m/stm32/l4/stm32l4.c

```diff
@@ -393,6 +467,15 @@ time_t _stm32_pwrEnterLPStop(time_t us)
 
 	*(stm32_common.scb + syst_csr) |= 1;
 
+	if (useMsi != 0) {
+		/* Restore pre-sleep MSI clock */
+		_stm32_rccHsiToMsi();
+	}
+
+	if (stm32_common.cpuclk <= 6 * 1000 * 1000) {
+		_stm32_pwrSetCPUVolt(2);
+	}
```

Comment:
With some small changes to _stm32_rccSetCPUClock it could be used in this function.

```
time_t _stm32_pwrEnterLPStop(time_t us) {
  time_t runClock = stm32_common.cpuclock;
  _stm32_rccSetCPUClock(HSI16_CLK);
  …
  _stm32_rccSetCPUClock(runClock);
}
```

This way voltage setting logic and clock source switching is done in single place.

Reply:
I was reluctant to change this API as it's a part of platformctl. At this point I avoid changes in system API

Reply:
Ok, now I understand what you're suggesting, although I'd rather not. I don't like assuming that `_stm32_rccSetCPUClock(16 Mhz)` will select HSI.